src/app/pages/listings/contents/service/listings-api.service.ts

```diff
@@ -0,0 +1,21 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ListingCategory, ListingsReponse } from 'src/app/models/listings';
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ListingsApiService {
+
+  constructor(private http:HttpClient) { }
+
+  getListings(category?:ListingCategory): Observable<ListingsReponse>{
+    const categoryParam = new HttpParams().set('category', category || '').toString();
+    return this.http.get<ListingsReponse>(`https://dev.to/search/listings?${categoryParam}&listing_search=&page=0&per_page=75&tag_boolean_mode=all`);
```

Comment:
you should use enviornment.ts file for https://dev.to/

Reply:
Currently in environment.ts we have api url as https://dev.to/api, can i add one more and refer it from there?

Reply:
Yeah add one `https://dev.to/`

Reply:
Rn I'm merging your pr you can do that in next pr